src/core/credrank/testUtils.js

```diff
@@ -158,15 +158,43 @@ export const weightedGraph: () => WeightedGraph = () => ({
   weights: weights(),
   graph: graph(),
 });
-export const args: () => MarkovProcessGraphArguments = () => ({
+export const args = (
+  personalAttributions?: PersonalAttributions = []
+): MarkovProcessGraphArguments => ({
   weightedGraph: weightedGraph(),
   parameters,
   intervals,
   participants: [participant1, participant2],
-  personalAttributions: [],
+  personalAttributions,
 });
-export const markovProcessGraph: () => MarkovProcessGraph = () =>
-  MarkovProcessGraph.new(args());
+const attribution1 = {
+  fromParticipantId: participant1.id,
+  recipients: [
+    {
+      toParticipantId: participant2.id,
+      proportions: [
+        {timestampMs: -2, proportionValue: 0.2},
+        {timestampMs: -1, proportionValue: 0.1},
```

Comment:
Seems a little strange to test with negative timestamps.

Reply:
Negative timestamps just represent time before January 1, 1970. They should be valid.